### ISSUE TITLE
Upon asking Doc DOT to search some code, it showed a source block and then immediately crashed.    I have been able to use Big Dot OK.   Some queries seem OK.  Some just crash.

### ISSUE THREAD

**Issue by Teravus** [alexpinel/Dot#26]
Upon asking Doc DOT to search some code, it showed a document source block and then immediately crashed.    I have been able to use Big Dot OK.   On Doc Dot, Some queries seem OK.  Some just crash, the app disappears and this ends up in the eventlog

Dot 0.9.3, installed using the windows installer.

Faulting application name: Dot.exe, version: 0.9.3.0, time stamp: 0x651560bd
Faulting module name: ucrtbase.dll, version: 10.0.26100.1882, time stamp: 0x52db7a2d
Exception code: 0xc0000409
Fault offset: 0x00000000000a502e
Faulting process id: 0x3424
Faulting application start time: 0x1DB4B574EB88CA8
Faulting application path: C:\Users\PERSON\AppData\Local\Programs\dot\Dot.exe
Faulting module path: C:\WINDOWS\System32\ucrtbase.dll
Report Id: 58a11b66-d17d-4fc9-b3c4-cf182c5972c6
Faulting package full name: 
Faulting package-relative application ID: 

Windows 11 Pro/24H2/13th Gen Intel(R) Core(TM) i9-13900H   2.60 GHz/32.0 GB (31.7 GB usable)
CPU mode  (Intel Xe Graphics)

Running dot as a standard user.
Python 3.10 standard environment, 
\>pip freeze
absl-py==2.1.0
astunparse==1.6.3
cachetools==5.5.0
certifi==2024.8.30
charset-normalizer==3.4.0
filelock==3.13.1
flatbuffers==24.3.25
fsspec==2024.6.1
gast==0.4.0
google-auth==2.36.0
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
grpcio==1.68.1
h5py==3.12.1
idna==3.10
Jinja2==3.1.4
keras==2.10.0
Keras-Preprocessing==1.1.2
libclang==18.1.1
Markdown==3.7
MarkupSafe==3.0.2
mpmath==1.3.0
networkx==3.3
numpy==2.2.0
oauthlib==3.2.2
opt_einsum==3.4.0
packaging==24.2
protobuf==3.19.6
pyasn1==0.6.1
pyasn1_modules==0.4.1
requests==2.32.3
requests-oauthlib==2.0.0
rsa==4.9
six==1.17.0
sympy==1.13.1
tensorboard==2.10.1
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorflow-cpu==2.10.0
tensorflow-directml-plugin==0.4.0.dev230202
tensorflow-estimator==2.10.0
tensorflow-io-gcs-filesystem==0.31.0
tensorflow_intel==2.10.0
termcolor==2.5.0
torch==2.5.1+xpu
typing_extensions==4.12.2
urllib3==2.2.3
Werkzeug==3.1.3
wrapt==1.17.0

npm --version
10.9.2
choco --version
2.4.1

**Comment by Teravus** [comment]
As a precaution, I'm re-installing the c++ redistributables, since, sometimes this is tied to redistributables, sometimes it is tied to the developer using different versions of the the windows SDK between steps.

**Comment by alexpinel** [comment]
Hi! Sorry to hear you are facing some issues. 

If Doc Dot sometimes works and sometimes it doesn't, it might be related to the number of sources it uses to answer questions. If for example, the context window is set as 4000 tokens and the chunk length is also set at 4000, Dot will only be able to support one chunk at a time (that is 1 source). If on the other hand chunk length is 2000, then Dot should be able to handle 2 sources. 

Doc Dot also uses a bit more RAM by default which could also cause some problems.

Please let me know if this helps and if you are facing other issues!

Alex

**Comment by Teravus** [comment]
Redistributable re-install/repair didn't help. 

With the default settings, when I can get it to work, it doesn't look particularly weighty
![image](https://github.com/user-attachments/assets/67076da3-89b7-4cef-b1b8-bffc18ce71cb)

It is affected by the settings. 
![image](https://github.com/user-attachments/assets/3ee5ac02-7a04-4377-8721-e1e6123b7b13)

Some queries just crash regardless.    "Summarize [document]"      Crash.

**Comment by Teravus** [comment]
That's all the time that I can spare on this right now.   If you don't find this useful, you can close this.
